src/test/java/com/wikia/webdriver/common/logging/VelocityWrapper.java

```diff
@@ -0,0 +1,204 @@
+package com.wikia.webdriver.common.logging;
+
+import static org.testng.internal.Utils.escapeHtml;
+
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+
+public class VelocityWrapper {
+
+  private static final String LAST_LOG_ROW_TEMPLATE_PATH = "lastLogRow.vm";
+  private static final String LOG_ROW_TEMPLATE_PATH = "logRow.vm";
+  private static final String LOG_ROW_WITH_LINK_TEMPLATE_PATH = "logRowWithLink.vm";
+  private static final String IMAGE_TEMPLATE_PATH = "image.vm";
+  private static final String ERROR_LOG_ROW_TEMPLATE_PATH = "errorLogRow.vm";
+  private static final String LOG_ROW_WITH_SCREENSHOT_TEMPLATE_PATH = "logRowWithScreenshot.vm";
+  private static final String LINK_TEMPLATE_PATH = "link.vm";
+  private static final String FIRST_LOG_ROW_TEMPLATE_PATH = "firstLogRow.vm";
+  private static final String BUTTON_TEMPLATE_PATH = "button.vm";
+  private static final String HEADER_TEMPLATE_PATH = "header.vm";
+  private static final String ERROR_LOG_ROW_WO_SCREENSHOT_AND_SOURCE_TEMPLATE_PATH =
+      "errorLogRowWoScreenshotAndSource.vm";
+  private static VelocityEngine velocityEngine = getVelocityEngine();
+
+  private VelocityWrapper() {
+    throw new IllegalAccessError("Utility class");
+  }
+
+  static String fillLogRow(List<String> classList, String command, String description) {
+    StringBuilder builder = new StringBuilder();
+
+    Template t = velocityEngine.getTemplate(LOG_ROW_TEMPLATE_PATH);
+    VelocityContext context = new VelocityContext();
+
+    context.put("className", "\"" + classList.stream().collect(Collectors.joining(" ")) + "\"");
+    context.put("command", command);
+    context.put("description", description);
+    StringWriter writer = new StringWriter();
+    t.merge(context, writer);
+    builder.append(writer.toString());
+    return builder.toString();
+  }
+
+  static String fillLastLogRow() {
+    StringBuilder builder = new StringBuilder();
+
+    Template t = velocityEngine.getTemplate(LAST_LOG_ROW_TEMPLATE_PATH);
+    VelocityContext context = new VelocityContext();
+
+    StringWriter writer = new StringWriter();
+    t.merge(context, writer);
+    builder.append(writer.toString());
+    return builder.toString();
+  }
+
+  public static String fillLogRowWithLink(String link, String label) {
+    StringBuilder builder = new StringBuilder();
+
+    Template t = velocityEngine.getTemplate(LOG_ROW_WITH_LINK_TEMPLATE_PATH);
+    VelocityContext context = new VelocityContext();
+    context.put("link", link);
+    context.put("label", label);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "label" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)